csrc/scheduler/normalization_inner.cpp

```diff
@@ -314,6 +344,20 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
         index_type);
   }
 
+  // Define two free parameters used in this heuristic.
+  // register_overhead is all registers except those for the persistent
+  // buffers. The register in each thread = register_overhead +
+  // persistent_buffer_size / bytes_per_register
+  // Current values are based on tests of sofmax, layer_norm, softmax_dropout,
+  // dropout_layer_norm on A100 & H100. It directly affects maxregcount passed
+  // to NVRTC and influences the occupancy.
+  const int64_t register_overhead = 16l;
```

Comment:
24 leads to regression in range (20K,24K), take 22K for example, vect = 8, batch = 6, threads_per_block = 480, min_reg_per_thread = 6*8 + 24 = 72, then warps_per_sm = 15. To reach 30 warps per sm, each thread can use only 64 registers, the corresponding overhead is 64 - 48 = 16. 
I also ran the perf test, same as expected, result is added to the doc.

Reply:
You are right, there is no reason to stick with a power of 2. I tested `24` on A100, overall it is slightly better than `16` or `32`, results are added to the doc.
Waiting for an avilable node to test H100.

Reply:
Excited to see the H100 number. I'll be a lot more comfortable with pulling the trigger for this change if it's a universal improvement.

Reply:
> Excited to see the H100 number. I'll be a lot more comfortable with pulling the trigger for this change if it's a universal improvement.

added results on H100. There is an error in the doc. I corrected it and it turns out that we don't need to change the register overhead for softmax. `32` is still the best value.

Reply:
Added H100 results. There is an error in the original doc. After comparing with the correct baseline data, we should still use `32` as the register overhead for softmax to avoid regressions.

Reply:
so looking at the graph we have earlier, if I'm reading this correctly, keeping the same register_overhead config means that the code change here gives performance boost across the board without any regression. Even though we are missing some spotty performance gain from a lowered registter_overhead heuristics.

I'm wondering if the regression is just because our current block size isn't optimal for certain persistent batch size. Let's put a TODO on this and revisit the topic after our step 2 in the design is done.

Reply:
You are right! The missing speedup will be addressed in our step-2.